tests/periph_uart_mode/main.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2018 Sparkmeter
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Manual test application for UART mode on CPUs with only one UART
+ *
+ * @author      Ben Postman <ben.l.postman@gmail.com>
+ *
+ * @}
+ */
+
+
+
+/*
+ * This application is intended for testing the UART mode function
+ * on CPUs that only have 1 UART. The existing UART test does not
+ * switching UART configuration mode on the device being used for
+ * shell. This test will cycle through all permutations of UART modes
+ * and print the mode to STDOUT. The test is evaluated by hooking up
+ * a probe or logic analyzer and confirming that each string can be
+ * read correctly. The strings will be printed as such:
+ * <data-bits><parity-bits><stop-bits>
+ * Example: 6O1 would be 6 data bits, odd parity, 1 stop bit
+ * If a mode is not supported, then a message indicating that will
+ * be printed using configuration from the last mode);
+ */
+
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "periph/uart.h"
+#include "xtimer.h"
+
+/* UART_DEV is always 0 since this test is for device with 1 UART */
+#define _UART_DEV (UART_DEV(0))
+
+
+static void _print_status(int8_t status, uart_data_bits_t data_bits,
+                          uart_parity_t parity, uart_stop_bits_t stop_bits) {
+    char mode[4];
+
+    /* If status < 0, print a header indicating an error */
+    if (status < 0) {
+        printf("Unsupported Mode: ");
```

Comment:
Could you make it not print anything if it is supported, it becomes pretty hard to read on the scope.

Reply:
Definitely, you mean if unsupported, right?

Reply:
Yes, skip (meaning don't print anything and don't wait any time) if the mode is unsupported.